app/scraper/__init__.py

```diff
@@ -58,74 +63,79 @@ def scrape(caches_active, face_book_cookie, people_search_session_cookie, csrf_t
+            for key, value in caches_active.items()
+        }
+        logger.info('Launching scraper.')
+        cache = Cache(caches_active)
+
+        cache_key = 'scraped_data'
+        logger.info('Checking cache...')
+        people = cache.get(cache_key)
+        face_book = None
+        if people:
+            logger.info('Found people in cache.')
+        else:
+            logger.info('Initializing sources.')
+            directory = sources.Directory(
+                cache, people_search_session_cookie, csrf_token)
+
+            print("facebook: ", cache, face_book_cookie, directory)
+            face_book = sources.FaceBook(
+                cache, face_book_cookie, directory)
+            name_coach = sources.NameCoach(
+                cache, people_search_session_cookie, csrf_token)
+            departmental = sources.Departmental(cache)
+
+            logger.info('Beginning scrape.')
+            people = []
+            thread_fb_dir_nc = Thread(target=scrape_face_book_directory_name_coach,
+                                      args=(face_book, directory, name_coach))
+            thread_departmental = Thread(
+                target=departmental.pull, args=(people,))
+            thread_fb_dir_nc.start()
+            thread_departmental.start()
+            thread_fb_dir_nc.join()
+            thread_departmental.join()
+            # TODO: find a cleaner way to exchange this data
+            people = name_coach.people
+            people = departmental.integrate(people)
+            cache.set(cache_key, people)
+
+        # Store people into database
+        logger.info('Inserting new data.')
+
+        # TODO: we do this at the starting of YaleConnect.merge; I'm just temporarily adding this so that the Person models can be deleted. Maybe we should make the deletes cascade or something?
+        db.session.query(leaderships).delete()
+        Group.query.delete()
+
+        Person.query.delete()
+        elasticsearch.indices.delete(index=Person.__tablename__)
+        elasticsearch.indices.create(index=Person.__tablename__)
+        num_inserted = 0
+        for person_dict in people:
+            if not person_dict.get('netid'):
+                continue
+            db.session.add(Person(**person_dict))
+            # Avoid memory overflows
+            num_inserted += 1
+            if num_inserted % 64 == 0:
+                db.session.commit()
+        db.session.commit()
+
+        # TODO: merge this into main scraper section;
+        # currently we do it after the rest of the scraper because
+        yaleconnect = sources.YaleConnect(cache, yaleconnect_cookie)
+        yaleconnect.pull(people)
+        yaleconnect.merge(people)
+
+        if face_book is not None and has_s3_credentials():
+            logger.info('Deleting unused images from S3.')
+            face_book.delete_unused_imgs(people)
+        logger.info('Done.')
+        # except Exception as e:
```

Comment:
This is here for a reason, to email the admins if the scraper fails. Could you uncomment?

Reply:
Done